Sources/Kitura/RouterResponse.swift

```diff
@@ -549,14 +425,173 @@ public class RouterResponse {
+            return self
+        }
+        guard let str = str else {
+            Log.warning("RouterResponse send(str:) invoked with a nil value")
+            return send("")
+        }
+        return send(str)
+    }
+    
+    /// Send the HTTP status code.
+    ///
+    /// - Parameter status: The HTTP status code.
+    /// - Returns: This RouterResponse.
+    public func send(status: HTTPStatusCode) -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(status:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        self.status(status)
+        send(HTTPURLResponse.localizedString(forStatusCode: status.rawValue))
+        return self
+    }
+    
+    // MARK: Send Data
+    
+    /// Send data.
+    ///
+    /// - Parameter data: The data to send.
+    /// - Returns: This RouterResponse.
+    @discardableResult
+    public func send(data: Data) -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(data:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        buffer.append(data: data)
+        state.invokedSend = true
+        return self
+    }
+    
+    /// Send a file.
+    ///
+    /// - Parameter fileName: The name of the file to send.
+    /// - Throws: An error in the Cocoa domain, if the file cannot be read.
+    /// - Returns: This RouterResponse.
+    ///
+    /// - Note: Sets the Content-Type header based on the "extension" of the file.
+    ///       If the fileName is relative, it is relative to the current directory.
+    @discardableResult
+    public func send(fileName: String) throws -> RouterResponse {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(fileName:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        let data = try Data(contentsOf: URL(fileURLWithPath: fileName))
+        
+        let contentType = ContentType.sharedInstance.getContentType(forFileName: fileName)
+        if  let contentType = contentType {
+            headers["Content-Type"] = contentType
+        }
+        
+        send(data: data)
+        
+        return self
+    }
+    
+    /// Set headers and attach file for downloading.
+    ///
+    /// - Parameter download: The file to download.
+    /// - Throws: An error in the Cocoa domain, if the file cannot be read.
+    public func send(download: String) throws {
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(download:) invoked after end() for \(self.request.urlURL)")
+            return
+        }
+        try send(fileName: StaticFileServer.ResourcePathHandler.getAbsolutePath(for: download))
+        headers.addAttachment(for: download)
+    }
+    
+    typealias JSONSerializationType = JSONSerialization
+    
+    /// Send JSON.
```

Comment:
Could we be more helpful here?  Something more like:
> Send structured data encoded in JSON format, and sets the Content-Type header appropriately.  The data must have a valid JSON structure, otherwise the data will not be sent and a warning will be logged.